### DISCUSSION TITLE
Poetry failing to run async python script by [tool.poetry.scripts] config

### DISCUSSION THREAD

**Discussion by jiaulislam** [python-poetry/poetry#6580] [Q&A] [answered]
Hi, I'm learning python asyncio module with a created project with Poetry. Poetry is able run the script when I run with `poetry run python main.py`  but whenever I try to run with  `pyproject.toml` file configured scripts way (`poetry run godzilla`), its giving me below error. Python Script and configs are given below. Is there any way I can run the project with  `pyproject.toml` configured way?
```
<string>:1: RuntimeWarning: coroutine 'main' was never awaited
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```
### Environment
- poetry - latest
- OS - UBUNTU+Windows
- Python 3.10

### pyproject.toml
```
[tool.poetry.scripts]
godzilla = "godzilla.main:main"
```
### main.py
```
import asyncio
import httpx
from devtools import debug


async def make_pokemon_request() -> dict:
    url = f"https://pokeapi.co/api/v2/pokemon/1"
    async with httpx.AsyncClient() as client:
        response: dict = await client.get(url)
        return response.json()


async def main() -> int:
    pokemon: dict = await make_pokemon_request()
    return pokemon.get("weight")


if __name__ == "__main__":
    weight = asyncio.run(main())
    debug(weight)

```

**Comment by clintonroy** [comment, marked as answer]
This is not a poetry problem.

Basically, the code you've got in your __name__ == "__main__" stanza, needs to be put into a function of it's own, so that it's run in an event loop.

So put that code in a function (call it, say, run) then change the tool.poetry.scripts entry to point to the run function.

**Reply by jiaulislam** [reply]
@clintonroy thanks a lot ! you saved me ❤
